ui/src/pages/Edit/UploadStory.tsx

```diff
@@ -0,0 +1,721 @@
+                rows={4}
+                id="author-bio"
+                name="bio"
+                label="Enter author biography"
+                defaultValue={formInput.bio}
+              />
+            </StyledBackgroundColor>
+
+            <StyledBackgroundColor>
+              <UploadStoriesHeading>
+                Additional Information
+              </UploadStoriesHeading>
+              <FormControl>
+                <UploadLabelsText>Country of Origin</UploadLabelsText>
+                <StyledAutocomplete
+                  color="Primary"
+                  loading={!countries}
+                  options={countries || []}
+                  forcePopupIcon
+                  name="author_country"
+                  id="select-label-country"
+                  freeSolo
+                  // if user adds a new country display it as the default value, if not, look for prefilled value (edit stories)
+                  // prettier-ignore
+                  value={state.addedCountry != "" ? state.addedCountry : (state.authorCountry ? state.authorCountry : null)}
+                  onInputChange={(_, newValue) =>
+                    dispatch({
+                      type: "SET_AUTOCOMPLETE_COUNTRY",
+                      autocompleteCountry: newValue,
+                    })
+                  }
+                  onChange={(_, newValue) => {
+                    setFormInput({
+                      ["author_country"]: newValue,
+                    });
+                    dispatch({ type: "SET_AUTHOR_COUNTRY", country: newValue });
+                  }}
+                  renderInput={(params) => {
+                    return (
+                      <TextField
+                        {...params}
+                        required
+                        variant="outlined"
+                        label=""
+                        placeholder="Enter story&#39;s country of origin *"
+                      />
+                    );
+                  }}
+                  PaperComponent={addNewCountry}
+                />
+              </FormControl>
+              <UploadLabelsText>Current City</UploadLabelsText>
+              <StyledAutocomplete
+                color="Primary"
+                loading={!citiesList}
+                options={citiesList || []}
+                ListboxComponent={
+                  ListboxComponent as React.ComponentType<
+                    React.HTMLAttributes<HTMLElement>
+                  >
+                }
+                renderGroup={renderGroup}
+                forcePopupIcon
+                freeSolo
+                name="current_city"
+                id="select-label-city"
+                value={state.currentCity != "" ? state.currentCity : null}
+                onChange={(_, newValue) => {
+                  setFormInput({
+                    ["current_city"]: newValue,
+                  });
+                  dispatch({ type: "SET_CURRENT_CITY", city: newValue });
```

Comment:
Maybe we can combine these into one big reducer state?

Reply:
I'm not sure what you mean by this. Do u mean refactoring the formInput reducer into the main reducer component?

Reply:
yup

Reply:
Okay so I tried going at this for a bit, but I'm having a tough time making it work. Here's what I'm doing right now: 
- created a formInput type in types.ts, and then added `formInput` as one of the fields in the State interface of my reducer. 
- working on switching the `setFormInput[]` methods in UploadStory to dispatches to the one reducer 

Where I'm stuck: 
let's say I want to change just the `Title` field of my formInput. Inside the case statements of my reducer, I have cases such as: 
```
case "title": {
      return{
        ...state,
        formInput.title: action.formData,
      }
    }
```

But this gives me errors as it doesn't let me edit formInput this way. 

pls lmk if this is the right approach, or if there's a better way to do this. Sorry for the long comment!